src/charm.py

```diff
@@ -108,6 +112,30 @@ def _configure(self):
 
         self.unit.status = ActiveStatus()
 
+    def _set_alerts(self, container):
+        """Create alert rule files for all Prometheus consumers.
+
+        Args:
+            container: the Prometheus workload container into which
+                alert rule files need to be created.
+        """
+        if not self.provider_ready:
+            return
+
+        container.remove_path(RULES_DIR, recursive=True)
```

Comment:
shouldn't the rules folder be on a persistent storage and the auto generated rules (a subfolder) within that?

Reply:
iirc remove_path can also raise ConnectionError

Reply:
It can, but you could use

```python
with container.is_ready():
    # do stuff
````

to handle that case...

Reply:
Yes it can. I am not a huge fan of littering my code with lot of Try Catch blocks and neither of using a try catch block for doing more than one thing. Alerts are pushed to workload containers after the Prometheus configuration itself is. If it happens that Prometheus configuration push succeeded and immediately after that alert failed it would be very very odd. But I will give this another thought.

Reply:
Thanks. switched to using `is_ready`.